### DISCUSSION TITLE
Disable individual prettier options

### DISCUSSION THREAD

**Discussion by caribou-code** [prettier/prettier#13925] [Q&A] [answered]
My understanding is that prettier sets defaults for options, for example I believe the `arrowParens` option defaults to `avoid`. Is there a way to disable specific options/rules entirely? For example if I don't want prettier to change arrow function param styling at all, so I want to disable the `arrowParens` rule entirely?

My main reason for this is because I can get ESLint to do everything I want from prettier except the `printWidth` option (the equivalent `max-len` rule in ESLint isn't compatible with autofix). So I want to disable all prettier rules except `printWidth`, so ESLint is my single source of truth for linting rules (with that 1 exception).

**Comment by kachkaev** [comment, marked as answer]
Hi @caribou-code! Disabling Prettier rules is not possible by design. Prettier works as a printer, so applies to the whole doc. The only option to make formatting exceptions is to use [`prettier-ignore` comments](https://prettier.io/docs/en/ignore.html). However, they are supposed to be used _very_ rarely.

See https://prettier.io/docs/en/comparison.html for more details.

The best option to make Prettier compatible with ESLint is to add [`eslint-config-prettier`](https://www.npmjs.com/package/eslint-config-prettier) to your project. It will turn conflicting ESLint rules _off_, thus making ESLint focused only on semantics. This works really well because formatting everything with Prettier instead of ESLint reduces the number of squiggly red lines you see in your editor. This reduces distraction. There are [other alternatives to combine ESLint and Prettier listed in docs](https://prettier.io/docs/en/related-projects.html#eslint-integrations), but I strongly recommend the first option. It may need a small shift in your mindset, but is definitely worth it long term!

**Reply by kachkaev** [reply]
A few more thoughts can be found here: https://github.com/prettier/eslint-plugin-prettier/issues/532#issuecomment-1344515008

**Reply by caribou-code** [reply]
Thanks, the solution you suggest is what I've been doing for the last few years, but the problem is needing to disable a large number of ESLint rules via `eslint-config-prettier`, needing to remember not to re-add any of those specific rules, and needing to get the order of ESLint extensions correct (Prettier goes last). It all seems very flakey.

I understand Prettier applies to the whole document, but why would it not be possible to disable some rules? There's already a `.prettierrc` config file to change individual rules. Couldn't we just set rules to `false` or `'off'` and those specific ones wouldn't be processed on the whole document?
